BlueprintUICommonControls/Sources/TransitionContainer.swift

```diff
@@ -29,3 +37,38 @@ public struct TransitionContainer: Element {
     }
 
 }
+
+public extension Element {
+
+    /// Wraps the element in a transition container to provide an animated transition.
+    ///
+    /// - Parameters:
+    ///   - onAppear: The transition to use when the element appears. By default, `.none`.
+    ///   - onDisappear: The transition to use when the element disappears. By default, `.none`.
+    ///   - onLayout: The animation to use when the element changes layout. By default, `.none`.
+    func transition(
+        onAppear: VisibilityTransition = .none,
+        onDisappear: VisibilityTransition = .none,
+        onLayout: LayoutTransition = .none
```

Comment:
I'm not sure these are the quite the defaults we want. In particular, the `.none` layout transition explicitly prevents inherited animations from taking effect on this element or its children.

It might be better to match the defaults on `ViewDescription` (`nil` for visibility, `inherited` for layout), so that not specifying one of these parameters will get you the same behavior as if you didn't have a TransitionContainer at all.

Reply:
Ahh shit, you're right. I forgot about this distinction before merging.

Reply:
@nononoah: Mind PRing this change? If not, I can do it since I did merge this.

Reply:
I have some thoughts on taking this a little further, which I'm drafting in #158.

Reply:
In the discussion, I mentioned that I was unsure of the direction of `.none`, but I was also unaware of its potential interaction with inherited animations.

If I understand your point correctly, any transition will override inherited transitions. This is reasonable behavior, and it should probably be documented in `TransitionContainer`. 

#158 looks like the alternative I first wrote up, but discarded as it was more disruptive. Please let me know if there's anything else I can do to help short of reviewing the PR, which I may not have permission to do.

Reply:
@nononoah No worries! I'll try to write a better explanation of the changes I'm suggesting in #158.